xradar/io/backends/nexrad_level2.py

```diff
@@ -0,0 +1,1381 @@
+        lat, lon, alt = get_nexrad_location(station)
+    elif (
+        "icao" in nfile.volume_header.keys()
+        and nfile.volume_header["icao"].decode()[0] == "T"
+    ):
+        lat, lon, alt = get_nexrad_location(nfile.volume_header["icao"].decode())
+    else:
+        lat, lon, alt = nfile.location()
+    latitude["data"] = np.array(lat, dtype="float64")
+    longitude["data"] = np.array(lon, dtype="float64")
+    altitude["data"] = np.array(alt, dtype="float64")
+
+    # Sweep information
+    sweep_number = {
+        "units": "count",
+        "standard_name": "sweep_number",
+        "long_name": "Sweep number",
+    }
+
+    sweep_mode = {
+        "units": "unitless",
+        "standard_name": "sweep_mode",
+        "long_name": "Sweep mode",
+        "comment": 'Options are: "sector", "coplane", "rhi", "vertical_pointing", "idle", "azimuth_surveillance", "elevation_surveillance", "sunscan", "pointing", "manual_ppi", "manual_rhi"',
+    }
+    sweep_start_ray_index = {
+        "long_name": "Index of first ray in sweep, 0-based",
+        "units": "count",
+    }
+    sweep_end_ray_index = {
+        "long_name": "Index of last ray in sweep, 0-based",
+        "units": "count",
+    }
+
+    if scans is None:
+        nsweeps = int(nfile.nscans)
+    else:
+        nsweeps = len(scans)
+    sweep_number["data"] = np.arange(nsweeps, dtype="int32")
+    sweep_mode["data"] = np.array(nsweeps * ["azimuth_surveillance"], dtype="S")
+
+    rays_per_scan = [s["nrays"] for s in scan_info]
+    sweep_end_ray_index["data"] = np.cumsum(rays_per_scan, dtype="int32") - 1
+
+    rays_per_scan.insert(0, 0)
+    sweep_start_ray_index["data"] = np.cumsum(rays_per_scan[:-1], dtype="int32")
+
+    # azimuth, elevation, fixed_angle
+    azimuth = get_azimuth_attrs()
+    elevation = get_elevation_attrs()
+    fixed_angle = {
+        "long_name": "Target angle for sweep",
+        "units": "degrees",
+        "standard_name": "target_fixed_angle",
+    }
+
+    azimuth["data"] = nfile.get_azimuth_angles(scans)
+    elevation["data"] = nfile.get_elevation_angles(scans).astype("float32")
+    fixed_agl = []
+    for i in nfile.get_target_angles(scans):
+        if i > 180:
+            i = i - 360.0
+            warnings.warn(
+                "Fixed_angle(s) greater than 180 degrees present."
+                " Assuming angle to be negative so subtrating 360",
+                UserWarning,
+            )
+        else:
+            i = i
+        fixed_agl.append(i)
+    fixed_angles = np.array(fixed_agl, dtype="float32")
+    fixed_angle["data"] = fixed_angles
+
+    # fields
+    max_ngates = len(_range["data"])
+
+    fields = {}
+    for moment in available_moments:
+        dic = get_moment_attrs(nexrad_mapping[moment])
+        dic["_FillValue"] = -9999
+        if delay_field_loading:
+            dic = LazyLoadDict(dic)
+            data_call = _NEXRADLevel2StagedField(nfile, moment, max_ngates, scans)
```

Comment:
Here is the pain point, this will extract the moments of different scans (which might be on different range resolutions.